test/e2e/nsx_subnet_test.go

```diff
@@ -469,3 +471,60 @@ func assureSubnetPort(t *testing.T, ns, subnetPortName string) (res *v1alpha1.Su
 	require.NoError(t, err)
 	return
 }
+
+func NoIPSubnet(t *testing.T) {
+	noIPSubnetPath, _ := filepath.Abs("./manifest/testSubnet/subnet-no-ip.yaml")
+	require.NoError(t, applyYAML(noIPSubnetPath, subnetTestNamespace))
+	defer deleteYAML(noIPSubnetPath, subnetTestNamespace)
+
+	noIPPortPath, _ := filepath.Abs("./manifest/testSubnet/subnetport-in-no-ip-subnet.yaml")
+	require.NoError(t, applyYAML(noIPPortPath, subnetTestNamespace))
+	defer deleteYAML(noIPPortPath, subnetTestNamespace)
+
+	assureSubnetPort(t, subnetTestNamespace, "port-in-no-ip-subnet")
+
+	// Check the no IP Subnet.
+	subnet, err := testData.crdClientset.CrdV1alpha1().Subnets(subnetTestNamespace).Get(context.TODO(), "subnet-no-ip", v1.GetOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, subnet.Status, "No Status info in Subnet")
+
+	// Check the no IP SubnetPort.
+	port, err := testData.crdClientset.CrdV1alpha1().SubnetPorts(subnetTestNamespace).Get(context.TODO(), "port-in-no-ip-subnet", v1.GetOptions{})
+	require.NoError(t, err)
```

Comment:
could you update the assureSubnetPort to reutrn port? since assureSubnetPort has the same logic for L492-493?

Reply:
good idea, done